purchases/src/main/kotlin/com/revenuecat/purchases/PurchasesFactory.kt

```diff
@@ -0,0 +1,110 @@
+package com.revenuecat.purchases
+
+import android.Manifest
+import android.app.Application
+import android.content.Context
+import android.content.pm.PackageManager
+import android.preference.PreferenceManager
+import androidx.annotation.VisibleForTesting
+import com.revenuecat.purchases.common.AppConfig
+import com.revenuecat.purchases.common.Backend
+import com.revenuecat.purchases.common.BillingAbstract
+import com.revenuecat.purchases.common.Dispatcher
+import com.revenuecat.purchases.common.HTTPClient
+import com.revenuecat.purchases.common.PlatformInfo
+import com.revenuecat.purchases.common.caching.DeviceCache
+import com.revenuecat.purchases.common.networking.ETagManager
+import com.revenuecat.purchases.identity.IdentityManager
+import com.revenuecat.purchases.subscriberattributes.SubscriberAttributesManager
+import com.revenuecat.purchases.subscriberattributes.SubscriberAttributesPoster
+import com.revenuecat.purchases.subscriberattributes.caching.SubscriberAttributesCache
+import java.net.URL
+import java.util.concurrent.ExecutorService
+import java.util.concurrent.Executors
+
+internal class PurchasesFactory {
+
+    fun createPurchases(
+        configuration: PurchasesConfiguration,
+        platformInfo: PlatformInfo,
+        proxyURL: URL?
+    ): Purchases {
+        validateConfiguration(configuration)
+
+        with(configuration) {
+            val application = context.getApplication()
+            val appConfig = AppConfig(
+                context,
+                observerMode,
+                platformInfo,
+                proxyURL,
+                store,
+                dangerousSettings
+            )
+
+            val prefs = PreferenceManager.getDefaultSharedPreferences(application)
```

Comment:
To be able to test this more thoroughly, we would need to create factories for each of these dependencies. Considering there is not much more logic here, I thought the value of doing that now wasn't that big, but lmk if you think otherwise or have any other ideas.

Reply:
I agree with you.